bridge/evm/script/deploy_bridge.s.sol

```diff
@@ -22,88 +22,104 @@ contract DeployBridge is Script {
+            deployConfig.supportedTokens[2] = deployConfig.WETH;
+            deployConfig.supportedTokens[3] = address(USDC);
+            deployConfig.supportedTokens[4] = address(USDT);
         }
 
         // convert supported chains from uint256 to uint8[]
-        uint8[] memory supportedChainIDs = new uint8[](config.supportedChainIDs.length);
-        for (uint256 i; i < config.supportedChainIDs.length; i++) {
-            supportedChainIDs[i] = uint8(config.supportedChainIDs[i]);
+        uint8[] memory supportedChainIDs = new uint8[](deployConfig.supportedChainIDs.length);
+        for (uint256 i; i < deployConfig.supportedChainIDs.length; i++) {
+            supportedChainIDs[i] = uint8(deployConfig.supportedChainIDs[i]);
         }
 
-        // deploy bridge config
-
-        BridgeConfig bridgeConfig =
-            new BridgeConfig(uint8(config.sourceChainId), config.supportedTokens, supportedChainIDs);
-
-        // deploy Bridge Committee
+        // deploy Bridge Committee ===================================================================
 
         // convert committeeMembers stake from uint256 to uint16[]
-        uint16[] memory committeeMemberStake = new uint16[](config.committeeMemberStake.length);
-        for (uint256 i; i < config.committeeMemberStake.length; i++) {
-            committeeMemberStake[i] = uint16(config.committeeMemberStake[i]);
+        uint16[] memory committeeMemberStake =
+            new uint16[](deployConfig.committeeMemberStake.length);
+        for (uint256 i; i < deployConfig.committeeMemberStake.length; i++) {
+            committeeMemberStake[i] = uint16(deployConfig.committeeMemberStake[i]);
         }
 
         address bridgeCommittee = Upgrades.deployUUPSProxy(
             "BridgeCommittee.sol",
             abi.encodeCall(
                 BridgeCommittee.initialize,
                 (
-                    address(bridgeConfig),
-                    config.committeeMembers,
+                    deployConfig.committeeMembers,
                     committeeMemberStake,
-                    uint16(config.minCommitteeStakeRequired)
+                    uint16(deployConfig.minCommitteeStakeRequired)
+                )
+            )
+        );
+
+        // deploy bridge config =====================================================================
+
+        address bridgeConfig = Upgrades.deployUUPSProxy(
+            "BridgeConfig.sol",
+            abi.encodeCall(
+                BridgeConfig.initialize,
+                (
+                    address(bridgeCommittee),
+                    uint8(deployConfig.sourceChainId),
+                    deployConfig.supportedTokens,
+                    deployConfig.tokenPrices,
+                    supportedChainIDs
                 )
             )
         );
 
-        // deploy vault
+        // initialize config in the bridge committee
+        BridgeCommittee(bridgeCommittee).initializeConfig(address(bridgeConfig));
```

Comment:
is it a legit worry that someone could monitor new contracts and front run the config? or is it too hypothetical?

Reply:
All the deploy txs are broadcasted at the same time so it would be very hard to front run the config init.

However, it is technically possible. This is why before any assets are deposited, the config address should be verified to be correct.